### PR TITLE
src: allow Etc/GMT IANA timezones

### PR DESCRIPTION
**Pull request by ryzokuken**
Fixes: https://github.com/ryzokuken/draft-ryzokuken-datetime-extended/issues/10

/cc @mj1856

### CODE REVIEW

File: sources/sections/05-date-time-format.adoc [hunk 1]
```diff
@@ -106,9 +106,9 @@ time-numoffset = ("+" / "-") time-partial
 time-offset    = "Z" / time-numoffset
 time-full      = time-partial time-offset
 
-timezone-part  = (ALPHA / "." / "_") *13(ALPHA / "." / "_" / "-") ; but not "." or ".."
+timezone-part  = (ALPHA / "." / "_") *13(ALPHA / "." / "_" / "-" / "+") ; but not "." or ".."
 timezone-id    = timezone-part *("/" timezone-part)
-timezone       = "[" timezone-id "]"
+timezone       = "[" (timezone-id / "Etc/GMT" ("+"/"-") 1*2DIGIT) "]"
```
Comment (mj1856) [inline sources/sections/05-date-time-format.adoc:111]: Do we still need this change?

**Comment by ryzokuken** [comment]
@mj1856 PTAL now. This should work fine.

**Comment by ryzokuken** [comment]
Rebased.

**Comment by mj1856** [comment]
LGTM.  Thanks!

**Pull request merged**